include/mo.php

```diff
@@ -73,5 +89,50 @@ public function import_from_db( $lang ) {
 	 */
 	public function delete_entry( $string ) {
 		unset( $this->entries[ $string ] );
+		unset( self::$cache[ $this->get_header( 'Language' ) ][ $string ] );
+	}
+
+	/**
+	 * Adds an entry to the MO structure and caches it.
+	 *
+	 * @since 3.7
+	 *
+	 * @param Translation_Entry|array $entry The entry to add.
+	 * @return bool True if the entry was added, false otherwise.
+	 */
+	public function add_entry( $entry ) {
+		if ( is_array( $entry ) ) {
+			$entry = new Translation_Entry( $entry );
+		}
+
+		if ( parent::add_entry( $entry ) ) {
+			self::$cache[ $this->get_header( 'Language' ) ][ $entry->key() ] = $entry;
+
+			return true;
+		}
+
+		return false;
+	}
```

Comment:
`get_header()` returned value checked in [ba2fc10](https://github.com/polylang/polylang/pull/1570/commits/ba2fc10a8c10321ca319209648a888975cb4e127).
But I wouldn't bail out early if no language is found in the header.
Indeed, `PLL_MO` must be usable *without* calling `import_from_db()` first. This way it would behave like a simple `Translations` object, juste like before.